Manifold/Module+String.swift

```diff
@@ -0,0 +1,35 @@
+//  Copyright © 2015 Rob Rix. All rights reserved.
+
+extension Module {
+	public static var string: Module {
+		let List: Term = "List"
+		let cons: Term = "cons"
+		let `nil`: Term = "nil"
+
+		let String = Declaration("String",
+			type: .Type,
+			value: .Embedded(Swift.String.self))
+
+		let Character = Declaration("Character",
+			type: .Type,
+			value: .Embedded(Swift.Character.self))
+
+		func toTerm(characters: Swift.String.CharacterView) -> Term {
+			switch characters.first {
+			case let .Some(c):
+				return cons[Character.ref, .Embedded(c), toTerm(characters.dropFirst())]
+			default:
+				return `nil`[Character.ref]
+			}
+		}
+
+		let toList = Declaration("toList",
+			type: String.ref --> List[Character.ref],
+			value: () => { (string: Term) -> Term in
+				guard case let .Embedded(value as Swift.String, _, _) = string.out else { return ("toList" as Term)[string] }
+				return toTerm(value.characters)
```

Comment:
This is incorrect.

We evaluate this function when the value term is constructed, not when the defined `toList` term is evaluated. Therefore, it always fails the `guard` clause, produces itself in neutral form, and diverges during evaluation.